### PR TITLE
User/dev1 adit/rsjobs2

### PR DESCRIPTION
**Pull request by AditTulasi**
Added test cases for existing jobs cmdlets. Fixed few issues in Hydra Adapter.

### CODE REVIEW

File: src/ResourceManager/RecoveryServices.Backup/CmdletWarningAndErrorMessages.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ internal class Job
             public const string RefineFilters = "There are more than 1000 jobs for the filter combination you have provided. Kindly refine your filters to fetch the job you want.";
             public const string AllowedDateTimeRangeExceeded = "To filter should not be more than 30 days away from From filter.";
             public const string JobIdAndJobMismatch = "JobID and Job object provided don't match each other";
-            public const string ToShouldBeLessThanFrom = "To filter should not less than From filter.";
+            public const string ToShouldBeGreaterThanFrom = "To filter should not be less than From filter.";
```
Comment (sudhakar4171) [inline src/ResourceManager/RecoveryServices.Backup/CmdletWarningAndErrorMessages.cs]: can you remove this file and add all your messages into resources.resx file in models. we have added for policy/item there. if you add in this file, then the string is not localized.

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/JobConversions.cs [hunk 2]
```diff
@@ -152,8 +160,41 @@ private static AzureRmRecoveryServicesAzureVmJobErrorInfo GetPSAzureVmErrorInfo(
...
+                case BackupManagementType.AzureVM:
+                    return "AzureIaasVM";
+                default:
+                    throw new Exception("Invalid BackupManagementType provided: " + mgmtType);
+            }
+        }
+
+        public static string GetPSBackupManagementType(string jobType)
+        {
+            if (jobType == "AzureIaasVM")
```
Comment (sudhakar4171) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/JobConversions.cs]: I think there is one static class which has these values for jobType in hydra. pls try to use it or define constant

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.HydraAdapter/CommonHelpers.cs [hunk 3]
```diff
@@ -25,7 +25,7 @@ public static string GetDateTimeStringForService(DateTime date)
             // we have to use english culture because our user might be running 
             // PS in another culture and our service can't understand it.
             DateTimeFormatInfo dateFormat = new CultureInfo("en-US").DateTimeFormat;
-            return date.ToString("yyyy-MM-dd hh:mm:ss tt", dateFormat);
+            return date.ToUniversalTime().ToString("yyyy-MM-dd hh:mm:ss tt", dateFormat);
```
Comment (sudhakar4171) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.HydraAdapter/CommonHelpers.cs]: I have written some method like ConvertToUTCDateTime() in policyconversions.cs .. pls check if you can reuse/generalize it

File: src/ResourceManager/RecoveryServices.Backup/CmdletWarningAndErrorMessages.cs (hunk 1 above)
Comment (AditTulasi) [inline src/ResourceManager/RecoveryServices.Backup/CmdletWarningAndErrorMessages.cs]: Sure, I'll move these into resx files.

Don't we need to do the same for HelpMessages file too then? Don't they need any localization?

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/JobConversions.cs [hunk 4]
```diff
@@ -152,8 +160,41 @@ private static AzureRmRecoveryServicesAzureVmJobErrorInfo GetPSAzureVmErrorInfo(
...
+            return splitArr[splitArr.Length - 1];
+        }
+
+        #endregion
+
         #endregion
 
+        #region Enum translators
+
+        public static string GetJobTypeForService(BackupManagementType mgmtType)
```
Comment (sudhakar4171) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/JobConversions.cs]: do you want to add these methods inside hydrahelpers.cs file?

File: src/ResourceManager/RecoveryServices.Backup/RecoveryServicesBackupCmdletBase.cs [hunk 5]
```diff
@@ -119,5 +126,36 @@ public override void ExecuteCmdlet()
...
+            }
+            return result;
+        }
+
+        public BackUpOperationStatusResponse WaitForOperation(string url)
+        {
+            // using this directly because it doesn't matter which function we use.
+            // return type is same and currently we are using it in only two places.
+            // protected item and policy.
+            BackUpOperationStatusResponse response = HydraAdapter.GetProtectedItemOperationStatusByURL(url);
```
Comment (sudhakar4171) [inline src/ResourceManager/RecoveryServices.Backup/RecoveryServicesBackupCmdletBase.cs]: why hardcoded to GetProtectedItemOperationStatusByURL? take this method name as 'Action' input.

File: src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/JobConversions.cs (hunk 4 above)
Comment (AditTulasi) [inline src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Conversions/JobConversions.cs]: This is specific to jobs. I am not sure if others want to use this. I'll keep it in this file.

**Pull request merged**